pandas/tools/tests/test_concat.py

```diff
@@ -919,6 +920,37 @@ def test_concat_keys_with_none(self):
                           keys=['b', 'c', 'd', 'e'])
         tm.assert_frame_equal(result, expected)
 
+    def test_union_categorical(self):
+        # GH 13361
+        s = Categorical(list('abc'))
```

Comment:
add a tests that makes sure that the returned categories are in order of appearance in the union (you might have an existing one which checks this accidently, but make an explict one, and mark it)